x-pack/plugins/lens/public/heatmap_visualization/chart_component.tsx

```diff
@@ -321,6 +321,12 @@ export const HeatmapComponent: FC<HeatmapRenderProps> = ({
         showLegend={args.legend.isVisible}
         legendPosition={args.legend.position}
         debugState={window._echDebugStateFlag ?? false}
+        theme={{
+          ...chartTheme,
+          legend: {
+            labelOptions: { maxLines: args.legend.truncate ? args.legend?.maxLines ?? 1 : 0 },
```

Comment:
nitpick: so when the value is 0, that means we don't truncate? maybe instead we could just use undefined?

Reply:
No, actually 0 means that the value is not truncated. This is how it is implemented on es-charts side

Reply:
Ok, all good then!